### PR TITLE
Bybit spot only

### PR DESCRIPTION
**Pull request by orikalinski**

### CODE REVIEW

File: python/ccxt/bybit.py [hunk 1]
```diff
@@ -585,10 +612,99 @@ def get_positions(self, symbol=None):
...
+                'spot': True,
+                'margin': None,
+                'swap': False,
+                'future': False,
+                'option': False,
+                'active': active,
+                'contract': False,
+                'linear': None,
+                'inverse': None,
+                'taker': self.parse_number('0.001'),
```
Comment (giladcornix) [inline python/ccxt/bybit.py:673]: Not critic but we might change this to 0.001 as the parse number is float('0.001')

File: python/ccxt/bybit.py [hunk 2]
```diff
@@ -1010,54 +1064,17 @@ def fetch_ohlcv(self, symbol, timeframe='1m', since=None, limit=None, params={})
...
         base = None
         marketId = self.safe_string(trade, 'symbol')
         amount = self.safe_float_2(trade, 'qty', 'exec_qty')
+        if amount is None:
+            amount = self.safe_float(trade, 'orderQty')
         cost = self.safe_float(trade, 'exec_value')
-        price = self.safe_float_2(trade, 'price', 'exec_price')
+        price = self.safe_float_2(trade, 'exec_price', 'price')
+        if price is None:
+            price = self.safe_float(trade, 'orderPrice')
```
Comment (giladcornix) [inline python/ccxt/bybit.py:1077]: safe_float_n?

File: python/ccxt/bybit.py [hunk 3]
```diff
@@ -1010,54 +1064,17 @@ def fetch_ohlcv(self, symbol, timeframe='1m', since=None, limit=None, params={})
...
-        #     }
-        #
-        id = self.safe_string_2(trade, 'id', 'exec_id')
+        _id = self.safe_string_2(trade, 'id', 'exec_id')
         symbol = None
         base = None
         marketId = self.safe_string(trade, 'symbol')
         amount = self.safe_float_2(trade, 'qty', 'exec_qty')
+        if amount is None:
+            amount = self.safe_float(trade, 'orderQty')
```
Comment (giladcornix) [inline python/ccxt/bybit.py:1073]: safe_float_n?

File: python/ccxt/bybit.py [hunk 4]
```diff
@@ -1321,28 +1285,30 @@ def parse_order(self, order, market=None):
...
         fee = None
-        if feeCost is not None:
-            feeCost = abs(feeCost)
+        feeCostString = self.safe_string_2(order, 'cum_exec_fee', 'cumExecFee')
+        if feeCostString is not None:
+            feeCurrency = market['quote'] if market['linear'] else market['base']
             fee = {
-                'cost': feeCost,
-                'currency': base,
+                'cost': feeCostString,
```
Comment (giladcornix) [inline python/ccxt/bybit.py:1301]: is won't cause problems? not only it was not string it was also with abs

File: python/ccxt/bybit.py [hunk 5]
```diff
@@ -1374,84 +1341,96 @@ def fetch_order(self, id, symbol=None, params={}):
...
             # 'stop_order_id': id,  # one of stop_order_id or order_link_id is required for conditional orders
         }
-        method = 'privateLinearGetOrderSearch' if self.is_linear(symbol) else 'privateGetOrder'
-        stopOrderId = self.safe_string(params, 'stop_order_id')
-        if stopOrderId is None:
-            orderLinkId = self.safe_string(params, 'order_link_id')
-            if orderLinkId is None:
+        order_type = self.safe_string_lower(params, 'type')
+        params = self.omit(params, ["type"])
+        is_conditional = order_type == 'stop'
```
Comment (giladcornix) [inline python/ccxt/bybit.py:1346]: making sure, so as we said, we are not sending(and still won't) in params type == 'stop', right?

**Review by giladcornix — CHANGES_REQUESTED** [review]

File: python/ccxt/bybit.py [hunk 6]
```diff
@@ -1368,43 +1426,50 @@ def fetch_order(self, id, symbol=None, params={}):
 
         response = getattr(self, method)(self.extend(request, params))
         result = self.safe_value(response, 'result', {})
-        return self.parse_order(result, market)
+        parsed_order = self.handle_stop_execution_order(self.fetch_order, result, symbol, is_conditional)
+        if parsed_order:
```
Comment (giladcornix) [inline python/ccxt/bybit.py:1430]: So if a stop order you assume that you will always get the fee field?

File: python/ccxt/bybit.py [hunk 7]
```diff
@@ -1368,43 +1426,50 @@ def fetch_order(self, id, symbol=None, params={}):
...
         }
-        if (upperCaseType == 'LIMIT') or (upperCaseType == 'LIMIT_MAKER'):
+        if (type == 'market') and (side == 'buy'):
+            if self.options['createMarketBuyOrderRequiresPrice']:
+                if price is None:
+                    raise InvalidOrder(self.id + " market buy order requires price argument to calculate cost(total amount of quote currency to spend for buying, amount * price). To switch off self warning exception and specify cost in the amount argument, set .options['createMarketBuyOrderRequiresPrice'] = False. Make sure you know what you're doing.")
+                else:
+                    # for market buy it requires the amount of quote currency to spend
+                    request['orderQty'] = self.cost_to_precision(symbol, float(amount) * float(price))
+            else:
```
Comment (giladcornix) [inline python/ccxt/bybit.py:1457]: no need, afraid that someone will think of turning that off

File: python/ccxt/bybit.py [hunk 8]
```diff
@@ -1368,43 +1426,50 @@ def fetch_order(self, id, symbol=None, params={}):
...
+        if is_post_only:
             request['orderType'] = 'LIMIT_MAKER'
-        clientOrderId = self.safe_string_2(params, 'clientOrderId', 'orderLinkId')
-        if clientOrderId is not None:
-            request['orderLinkId'] = clientOrderId
-        params = self.omit(params, ['clientOrderId', 'orderLinkId', 'postOnly'])
+        client_order_id = self.safe_string(params, 'order_link_id')
+        if client_order_id is not None:
+            request['orderLinkId'] = client_order_id
+        params = self.omit(params, ['order_link_id', 'post_only'])
```
Comment (giladcornix) [inline python/ccxt/bybit.py:1472]: no post_only?

File: python/ccxt/bybit.py [hunk 9]
```diff
@@ -1684,14 +1749,22 @@ def cancel_order(self, id, symbol=None, params={}):
...
                 method = 'privatePostV2PrivateOrderCancel'
         elif self.is_spot():
-            request['orderId'] = id
+            order_link_id = self.safe_string(params, 'order_link_id')
+            if order_link_id:
+                request['orderLinkId'] = order_link_id
+                params = self.omit(params, "order_link_id")
+            else:
+                request['orderId'] = id
             if is_conditional:
```
Comment (giladcornix) [inline python/ccxt/bybit.py:1758]: are we sending the stop_order_id also in ByBit SPOT? cause if not it will always be orderCategory=0

Comment (giladcornix) [reply python/ccxt/bybit.py:1758]: thinking about this, wouldn't it be better to send "type" in here as well and be consistent with:
is_conditional = order_type == 'stop'

**Review by giladcornix — CHANGES_REQUESTED** [review]

File: python/ccxt/bybit.py (hunk 8 above)
Comment (orikalinski) [reply python/ccxt/bybit.py:1472]: it's being used inside the following function

is_post_only = self.is_post_only(upper_case_type == 'MARKET', type == 'LIMIT_MAKER', params)

**Pull request merged**